c10/core/DispatchKeySet.h

```diff
@@ -121,29 +120,81 @@ class DispatchKeySet final {
   constexpr DispatchKeySet(uint64_t repr) : repr_(repr) {}
   uint64_t repr_ = 0;
 
+};
+
+/* Note [RuntimeDispatchKeySet v.s. DispatchKeySet]
+ * RuntimeDispatchKeySet is an extended version of DispatchKeySet which additionally
+ * supports iterating and can include Undefined dispatch key.
+ * Here're some typical use cases:
+ * - Alias dispatch key maps to several runtime dispatch keys. We need to iterate
+ *   through the list when precomputing dispatchTable_.
+ * - Some alias dispatch keys e.g. Math/DefaultBackend, can map to Undefined dispatch
+ *   key which is not representable in a DispatchKeySet.
+ * - Undefined is included in autogradother_backends since it's also mapped to
+ *   AutogradOther.
+ *
+ * When to use RuntimeDispatchKeySet?
+ * Only use RuntimeDispatchKeySet when you need to iterate through the set or you need to
+ * check if Undefined is included. In all other use cases please either use DispatchKeySet
+ * directly or work on RuntimeDispatchKeySet::key_set() since DispatchKeySet is smaller in size.
+ * If you are thinking about adding new APIs to RuntimeDispatchKeySet other than key_set()/has(),
+ * first check whether it can be done on DispatchKeySet.
+ */
+class RuntimeDispatchKeySet final {
+public:
+  constexpr RuntimeDispatchKeySet()
+    : ks_(DispatchKeySet()), has_undefined_(false) {}
+  constexpr RuntimeDispatchKeySet(DispatchKey k)
+    : ks_(DispatchKeySet(k)), has_undefined_(k == DispatchKey::Undefined) {}
+  constexpr RuntimeDispatchKeySet(DispatchKeySet ks, bool has_undefined=false)
+    : ks_(ks), has_undefined_(has_undefined) {}
+  constexpr RuntimeDispatchKeySet(std::initializer_list<DispatchKey> ks)
+    : ks_(DispatchKeySet(ks)), has_undefined_(false)
+    {
+      for (DispatchKey k: ks) {
+        if (k == DispatchKey::Undefined) {
+          has_undefined_ = true;
+          break;
+        }
+      }
+    }
+
+  constexpr DispatchKeySet key_set() const {
+    return ks_;
```

Comment:
This method is dangerous because it encourages people to discard the "has undefined" data. You should probably be thinking hard if you actually need this converter or not.